[17.0] [ADD] web_font_size_report_layout

syci opened the pull request:
Odoo’s default report layout uses a fixed font size for all PDF documents.
In many real use cases, especially when invoice or sales order lines include long product descriptions or detailed content, this results in documents that are difficult to read or excessively long.

This module introduces a configurable font size in the Document Layout settings.
Administrators can choose between 9pt, 10pt, 12pt, and 14pt to optimize readability and space usage, adapting reports to company needs (compactness vs. visibility).

victoralmau committed:
[ADD] web_pwa_customize: New module

TT51658

Commit:
[UPD] Update web_pwa_customize.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

carlos-lopez-tecnativa committed:
[FIX] web_responsive: prevent error in tests when accessing an undefined variable

For an unknown reason, when running the tests, the session does not have all variables filled, so apps_menu is undefined.
This causes an error in tours of other modules: "Cannot read properties of undefined (reading 'theme')".

The curious thing is that if you run the tests of this module alone, they pass.
But when running tests together with another module, tests from the other module fail.
For example, when running tests with the voip_oca module.

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: web-17.0/web-17.0-web_pwa_customize
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_pwa_customize/it/

hbrunn committed:
[FIX] web_widget_x2many_2d_matrix: restore pre-v17 functionality (axes)

use components to render axes and aggregates, allowing custom widgets
there

carlos-lopez-tecnativa committed:
[IMP] web_widget_product_label_section_and_note: force a width on the column

Odoo does not facilitate inheritance in the function https://github.com/odoo/odoo/blob/3bc173a6b22f2e360ec363277800ee5efa12088f/addons/web/static/src/views/list/list_renderer.js#L406 so it is necessary to force a max width at this low level.

In v18, Odoo provides an attribute at the field level
https://github.com/odoo/odoo/blob/1702cb44ab7692d70bfabfc25db9592dac85e7a2/addons/web/static/src/views/list/column_width_hook.js#L299  and in this widget Odoo uses a default value, which we also use here, see https://github.com/odoo/odoo/blob/1702cb44ab7692d70bfabfc25db9592dac85e7a2/addons/account/static/src/components/product_label_section_and_note_field/product_label_section_and_note_field.js#L249

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update web_widget_x2many_2d_matrix.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-17.0/web-17.0-web_widget_x2many_2d_matrix
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_widget_x2many_2d_matrix/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: web-17.0/web-17.0-web_widget_x2many_2d_matrix
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_widget_x2many_2d_matrix/it/

ajfebres committed:
[IMP] Adds a font size selector (pt) to the Document Layout wizard

syci committed:
Merge pull request #1 from BinhexTeam/17.0-web_font_size_report_layout

[IMP] Adds a font size selector (pt) to the Document Layout wizard

syci committed:
Update README.rst

syci committed:
Update README.rst

syci committed:
[FIX] web_font_size_report_layout: readme

syci committed:
[ADD] web_font_size_report_layout: tests

syci committed:
[FIX] web_font_size_report_layout: tests

syci committed:
[FIX] web_font_size_report_layout: tests

syci committed:
[FIX] web_font_size_report_layout: tests

AnsarLatif commented:
Tested on Odoo 17 CE — font size controls work in report editor, PDFs render cleanly. Functionality as expected.

remi-filament committed:
Translated using Weblate (French)

Currently translated at 96.7% (30 of 31 strings)

Translation: web-17.0/web-17.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_responsive/fr/

ajfebres committed:
Merge remote-tracking branch 'upstream/17.0' into 17.0

syci committed:
Merge pull request #3 from BinhexTeam/17.0

17.0 update

syci commented:
ping @pedrobaeza

pedrobaeza commented:
Sorry,not interested in this module.

pedrobaeza commented:
Well, me not being interested doesn't mean that OCA is not interested. I was just saying that I'm not reviewing it because of that.

syci commented:
@pedrobaeza Don´t Worry, i will retry PR. My origin branch is bad

syci commented:
Re-open on #3371
